DeDRM_plugin/ion.py

```diff
@@ -815,18 +815,18 @@ def __init__(self, voucherenv, dsn, secret):
         addprottable(self.envelope)
 
     def decryptvoucher(self):
-        shared = "PIDv3" + self.encalgorithm + self.enctransformation + self.hashalgorithm
+        shared = ("PIDv3" + self.encalgorithm + self.enctransformation + self.hashalgorithm).encode('ASCII')
 
         self.lockparams.sort()
         for param in self.lockparams:
             if param == "ACCOUNT_SECRET":
-                shared += param + self.secret
+                shared += param.encode('ASCII') + self.secret
```

Comment:
https://github.com/apprenticeharper/DeDRM_tools/blob/45a1a64db59911c133b48a1425eee2ad6e7a5f2b/DeDRM_plugin/kfxdedrm.py#L69-L80

`self.dsn` and `self.secret` both are `str`, looks like this pr should be reverted. It's kind hard to figure out the type of parameters in this project, can we use [typing](https://docs.python.org/3/library/typing.html)?